tests/unit/client.py

```diff
@@ -13,140 +13,139 @@
 # governing permissions and limitations under the License.
 
 """Unit tests for client library"""
-from flask.ext.sqlalchemy import SQLAlchemy
 from hil.flaskapp import app
-from hil.model import NetworkingAction
 from hil.client.base import ClientBase, FailedAPICallException
-from hil.client.client import Client, RequestsHTTPClient
+from hil.client.client import Client, HTTPClient, HTTPResponse
+from hil.test_common import config_testsuite, config_merge, \
+    fresh_database, fail_on_log_warnings
+from hil.model import db
+from hil import config, deferred, server
 
 import json
-import os
 import pytest
-import requests
-import sys
-import tempfile
-import time
 
-from subprocess import check_call, Popen
+from urlparse import urlparse
+from base64 import urlsafe_b64encode
+from passlib.hash import sha512_crypt
 
-ep = "http://127.0.0.1:8000" or os.environ.get('HIL_ENDPOINT')
-username = "hil_user" or os.environ.get('HIL_USERNAME')
-password = "hil_pass1234" or os.environ.get('HIL_PASSWORD')
+ep = "http://127.0.0.1:8000"
+username = "hil_user"
+password = "hil_pass1234"
 
-http_client = RequestsHTTPClient()
-http_client.auth = (username, password)
+
+class FlaskHTTPClient(HTTPClient):
+
+    def __init__(self):
+        self._flask_client = app.test_client()
+
+    def request(self, method, url, data=None, params=None):
+
+        # Flask doesn't provide a straightforward way to do basic auth,
+        # but it's not actually that complicated:
+        auth_header = 'Basic ' + urlsafe_b64encode(username + ':' + password)
+
+        resp = self._flask_client.open(
+            method=method,
+            headers={'Authorization': auth_header},
+            # flask expects just a path, and assumes
+            # the host & scheme:
+            path=urlparse(url).path,
+            data=data,
+            query_string=params,
+        )
+        return HTTPResponse(status_code=resp.status_code,
+                            headers=resp.headers,
+                            content=resp.get_data())
+
+http_client = FlaskHTTPClient()
 C = Client(ep, http_client)  # Initializing client library
 MOCK_SWITCH_TYPE = 'http://schema.massopencloud.org/haas/v0/switches/mock'
 OBM_TYPE_MOCK = 'http://schema.massopencloud.org/haas/v0/obm/mock'
 OBM_TYPE_IPMI = 'http://schema.massopencloud.org/haas/v0/obm/ipmi'
 
```

Comment:
We aren't using these constants anywhere in the tests. Might as well remove it (or use it) while we are cleaning up the tests.